File: sample/src/main/java/com/mandria/android/mvp/example/di/annotations/PresenterKey.java
package com.mandria.android.mvp.example.di.annotations;

import com.mandria.android.mvp.Presenter;

import dagger.MapKey;

/** Annotation to ensure [MapKey] values are [ViewModel] extensions. */
@MapKey
public @interface PresenterKey {

    Class<? extends Presenter> value();
}

File: demo_dagger_kotlin/src/main/kotlin/com/mandria/android/mvp/kotlin/example/di/PresenterKey.kt
package com.mandria.android.mvp.kotlin.example.di

import com.mandria.android.mvp.Presenter
import dagger.MapKey
import kotlin.reflect.KClass


/** Annotation to ensure [MapKey] values are [ViewModel] extensions. */
@Target(
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER
)
@Retention(AnnotationRetention.RUNTIME)
@MapKey
annotation class PresenterKey(val value: KClass<out Presenter<*>>)